### PR TITLE
Use a lock to ensure it is thread safe.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge or @vekterli PR

### CODE REVIEW

File: storage/src/tests/persistence/common/persistenceproviderwrapper.h [hunk 1]
```diff
@@ -50,8 +51,10 @@ class PersistenceProviderWrapper : public spi::AbstractPersistenceProvider
 private:
     spi::PersistenceProvider& _spi;
     spi::Result _result;
+    mutable std::mutex  _lock;
     mutable std::vector<std::string> _log;
     uint32_t _failureMask;
+    using Guard = std::unique_lock<std::mutex>;
```
Comment (toregge) [inline storage/src/tests/persistence/common/persistenceproviderwrapper.h:57]: std::lock_guard&lt;std::mutex&gt; should be sufficient here.

File: storage/src/tests/persistence/common/persistenceproviderwrapper.h [hunk 2]
```diff
@@ -61,12 +64,16 @@ class PersistenceProviderWrapper : public spi::AbstractPersistenceProvider
      * return the given error without the wrapped SPI ever being invoked.
      */
     void setResult(const spi::Result& result) {
+        Guard guard(_lock);
         _result = result;
     }
     void clearResult() {
         _result = spi::Result(spi::Result::ErrorType::NONE, "");
     }
-    const spi::Result& getResult() const { return _result; }
```
Comment (toregge) [inline storage/src/tests/persistence/common/persistenceproviderwrapper.h:69]: clearResult() above doesn't seem to be used. Consider removing it.

**Review by toregge — APPROVED** [review]

File: storage/src/tests/persistence/common/persistenceproviderwrapper.h (hunk 1 above)
Comment (baldersheim) [reply storage/src/tests/persistence/common/persistenceproviderwrapper.h]: Thanks, fixed

File: storage/src/tests/persistence/common/persistenceproviderwrapper.h (hunk 2 above)
Comment (baldersheim) [reply storage/src/tests/persistence/common/persistenceproviderwrapper.h:69]: Good catch, done.

**Review by vekterli — COMMENTED** [review]
:+1:

**Pull request merged**